helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/PropertyStoreAccessor.java

```diff
@@ -134,4 +135,28 @@ public Response putPropertyByPath(@PathParam("clusterId") String clusterId,
       return serverError(e);
     }
   }
+
+  /**
+   * Recursively deletes the PropertyStore path. If the node does not exist, it returns OK().
+   * @param clusterId
+   * @param path
+   * @return
+   */
+  @DELETE
+  @Path("{path: .+}")
+  public Response deletePropertyByPath(@PathParam("clusterId") String clusterId,
+      @PathParam("path") String path) {
+    path = "/" + path;
+    if (!ZkValidationUtil.isPathValid(path)) {
+      LOG.info("The propertyStore path {} is invalid for cluster {}", path, clusterId);
+      return badRequest(
+          "Invalid path string. Valid path strings use slash as the directory separator and names the location of ZNode");
```

Comment:
Nit, long line? Maybe split it into two strings concatenation?

Reply:
This has been formatted with helix-style, so I don't think we need to do a string concatenation.

Reply:
helix-style won't be able to limit such long string within 100 chars per line. We just need to do it manually. If we have checkstyle for 100 chars check, this won't pass.

Reply:
If it's not something we could strictly enforce, then I don't see a strong reason for doing so. That is if we have a checkstyle check- we don't have this check.

If you want to enforce a lint/checkstyle check, then we could again do it project-wide.